photon/src/PhotonFeeder.cpp

```diff
@@ -131,46 +133,65 @@ void PhotonFeeder::identify() {
 void PhotonFeeder::showVersion() {
     delay(500);
 
-    if(VERSION.major == 0){ // flashing red, beta
+    if(_version.indexOf("debug") >= 0){ // flashing red, beta
         set_rgb(true, false, false);
         delay(250);
         set_rgb(false, false, false);
         delay(250);
     }
 
-    else if (VERSION.major == 1){ // flashing green, v1.X.X
+    else { // flashing green, v1.X.X
 
-        int flash = 0;
+        set_rgb(false, true, false);
+        delay(250);
+        set_rgb(false, false, false);
+        delay(250);
 
-        if (VERSION.patch == 3){
-            flash = 1;
-        }
-        else if (VERSION.patch == 4){
-            flash = 2;
-        }
+    }
+}
+
+void PhotonFeeder::vendorSpecific(uint8_t options[VENDOR_SPECIFIC_OPTIONS_LENGTH], uint8_t* response) {
+    uint8_t command = options[0];
+
+    int len = _version.length();
 
-        for(int i=0;i<flash;i++){
-            set_rgb(false, true, false);
-            delay(250);
-            set_rgb(false, false, false);
-            delay(250);
+    // 0x00 - 0x0F is Firmware Version, section by section.
+    // characters byte_0 times 20 and byte_0 + 1 times 20 is the range
+    if(command <= 0x0F) {
+
+        int startingIndex = command * VENDOR_SPECIFIC_OPTIONS_LENGTH;
+
+        for(int i = 0; i < VENDOR_SPECIFIC_OPTIONS_LENGTH; i++){
+            
+            if((startingIndex + i) < len){
+                char currentChar = _version.substring(startingIndex + i, startingIndex + i + 1)[0];
+                response[i] = static_cast<unsigned char>(currentChar);
+            }
+            else{
+                response[i] = 0x0;
+            }
```

Comment:
You can do this copying a bit more efficiently using `strncpy`.

```suggestion
    // Commands 0x00 - 0x0F get the firmware version string in chunks. The firmware version string may be smaller than
    // the allotted 7 chunks, but it is null-terminated as expected.
    if(command <= 0x0F) {
        const auto start_index = min(size_t(command) * VENDOR_SPECIFIC_OPTIONS_LENGTH, _version.length());
        const auto chunk = _version.c_str() + start_index;

        // NOTE: Using strncpy here specifically because we want its behavior where it handles encountering a null byte
        // in the src string.
        strncpy((char*)(response), chunk, VENDOR_SPECIFIC_OPTIONS_LENGTH);
    }
```